x-pack/plugins/security_solution/server/endpoint/services/actions/clients/crowdstrike/crowdstrike_actions_client.test.ts

```diff
@@ -0,0 +1,293 @@
+
+describe('CrowdstrikeActionsClient class', () => {
+  let classConstructorOptions: CrowdstrikeActionsClientOptionsMock;
+  let crowdstrikeActionsClient: ResponseActionsClient;
+  let connectorActionsMock: ActionsClientMock;
+
+  const createCrowdstrikeIsolationOptions = (
+    overrides: Omit<
+      Parameters<typeof responseActionsClientMock.createIsolateOptions>[0],
+      'agent_type'
+    > = {}
+  ) => responseActionsClientMock.createIsolateOptions({ ...overrides, agent_type: 'crowdstrike' });
+
+  beforeEach(() => {
+    classConstructorOptions = CrowdstrikeMock.createConstructorOptions();
+    connectorActionsMock = classConstructorOptions.connectorActions;
+    crowdstrikeActionsClient = new CrowdstrikeActionsClient(classConstructorOptions);
+  });
+
+  it.each([
+    'killProcess',
+    'suspendProcess',
+    'runningProcesses',
+    'getFile',
+    'execute',
+    'upload',
+  ] as Array<keyof ResponseActionsClient>)(
+    'should throw an un-supported error for %s',
+    async (methodName) => {
+      // @ts-expect-error Purposely passing in empty object for options
+      await expect(crowdstrikeActionsClient[methodName]({})).rejects.toBeInstanceOf(
+        ResponseActionsNotSupportedError
+      );
+    }
+  );
+
+  it('should error if unable to retrieve list of connectors', async () => {
+    connectorActionsMock.getAll.mockImplementation(async () => {
+      throw new Error('oh oh');
+    });
+    const responsePromise = crowdstrikeActionsClient.isolate(createCrowdstrikeIsolationOptions());
+
+    await expect(responsePromise).rejects.toBeInstanceOf(ResponseActionsClientError);
+    await expect(responsePromise).rejects.toHaveProperty(
+      'message',
+      expect.stringContaining('Unable to retrieve list of stack connectors:')
+    );
+    await expect(responsePromise).rejects.toHaveProperty('statusCode', 400);
+  });
+
+  it('should error if retrieving connectors fails', async () => {
+    (connectorActionsMock.getAll as jest.Mock).mockImplementation(async () => {
+      throw new Error('oh oh');
+    });
+
+    await expect(
+      crowdstrikeActionsClient.isolate(createCrowdstrikeIsolationOptions())
+    ).rejects.toMatchObject({
+      message: `Unable to retrieve list of stack connectors: oh oh`,
+      statusCode: 400,
+    });
+  });
+
+  it.each([
+    ['no connector defined', async () => []],
+    [
+      'deprecated connector',
+      async () => [responseActionsClientMock.createConnector({ isDeprecated: true })],
+    ],
+    [
+      'missing secrets',
+      async () => [responseActionsClientMock.createConnector({ isMissingSecrets: true })],
+    ],
+  ])('should error if: %s', async (_, getAllImplementation) => {
+    (connectorActionsMock.getAll as jest.Mock).mockImplementation(getAllImplementation);
+
+    await expect(
+      crowdstrikeActionsClient.isolate(createCrowdstrikeIsolationOptions())
+    ).rejects.toMatchObject({
+      message: `No Crowdstrike stack connector found`,
```

Comment:
You can delete these tests around interactions with the Connector. My [PR here](https://github.com/elastic/kibana/pull/179388) has moved all of that logic to a new Class (`NormalizedExternalConnectorClient`) and these types of tests are now covered there so there is no need to include them in a per-client basis (unless you were to introduce other code that `.catch()` them and applied further  logic)